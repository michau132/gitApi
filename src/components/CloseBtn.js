import React from 'react';
import styled from 'styled-components';
import { Link } from 'react-router-dom';

const CloseBtnStyles = styled.div`
  position: fixed;
  right: 40px;
  top: 25px;
  width: 32px;
  height: 32px;
  z-index: 5;
  cursor: pointer;
  background-color: #333;
  border-radius: 4px;
  
  :hover {
    opacity: 0.5;
  }
  :before, :after {
    content: ' ';
    position: absolute;
    left: 15px;
    height: 33px;
    width: 2px;
    background-color: #fff;
    border-radius: 15px;
  }
  :before {
    transform: rotate(45deg);
  }
  :after {
    transform: rotate(-45deg);
  }
  
  ${({ theme }) => theme.media.desktop} {
    position: absolute;
  }
`;

const CloseBtn = () => (
  <Link to={{ pathname: '/trainingGit/', state: { modal: false } }}>
    <CloseBtnStyles />
  </Link>

);

export default CloseBtn;
