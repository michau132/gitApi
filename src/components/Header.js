import React from 'react';
import styled from 'styled-components';

const H1 = styled.h1`
  font-size: 40px;
  color: #fdc22a;
  text-transform: uppercase;
  margin-left: 40px;
  span {
    color: #b6b6b6;
  }
`;

const Header = () => (
  <H1>
      beer
    <span>guru</span>
  </H1>
);

export default Header;
