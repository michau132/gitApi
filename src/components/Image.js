import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

const ImageStyles = styled.img`
  object-fit: contain;
  max-height: 100%;
  max-width: 100%;
  height: 100%;
  width: 100%;
`;

const Image = ({ src, alt }) => (
  <ImageStyles src={src} alt={alt} />
);

Image.propTypes = {
  src: PropTypes.string.isRequired,
  alt: PropTypes.string.isRequired,
};

export default Image;
