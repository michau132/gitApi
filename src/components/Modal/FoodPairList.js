import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import FoodPairListItem from './FoodPairListItem';

const FoodPairListStyle = styled.ul`
  list-style: none;
  padding: 0;
  margin-bottom: 10px;
  
  ${({ theme }) => theme.media.desktop} {
    margin-bottom: 20px;
  }
`;

const FoodPairList = ({ food_pairing }) => (
  <FoodPairListStyle>
    <h4>Best served with:</h4>
    {
      food_pairing.map(item => (
        <FoodPairListItem key={item} text={item} />
      ))
    }
  </FoodPairListStyle>
);

FoodPairList.propTypes = {
  food_pairing: PropTypes.arrayOf(PropTypes.string).isRequired,
};

export default FoodPairList;
