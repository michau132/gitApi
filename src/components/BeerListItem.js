import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import Image from './Image';

const BeerListItemStyle = styled.li`
  padding: 20px;
  background-color: #fff;
  box-shadow: 0px 0px 4px 1px rgba(122,122,122,0.9);
  transition: .5s all;
  text-align: center;
  margin: 20px;
  width: 320px;
  cursor: pointer;
  
  :hover {
    transform: scale(1.05);
    box-shadow: 0px 0px 29px 9px rgba(122,122,122,1);
  }
`;

const ImgWrapper = styled.div`
  height: 180px;
  margin-bottom: 10px;
`;

const BeerName = styled.div`
  font-size: 24px;
  color: ${({ theme }) => theme.color.title};
  font-weight: bold;
`;

const BeerDesc = styled.div`
  font-size: 18px;
  color: #b7b7b7;
`;

const BeerListItem = ({
  id, image_url, name, tagline, handleClick,
}) => (
  <BeerListItemStyle
    onClick={() => handleClick(id)}
  >
    <div>
      <ImgWrapper>
        <Image src={image_url} alt={name} />
      </ImgWrapper>
    </div>
    <BeerName>{name}</BeerName>
    <BeerDesc>
      {tagline}
    </BeerDesc>
  </BeerListItemStyle>
);

BeerListItem.propTypes = {
  id: PropTypes.number.isRequired,
  image_url: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
  tagline: PropTypes.string.isRequired,
  handleClick: PropTypes.func.isRequired,
};

export default BeerListItem;
