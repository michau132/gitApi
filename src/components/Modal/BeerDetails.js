import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import Image from '../Image';

const mediaTablet = ({ theme }) => theme.media.tablet;

const BeerDetailsCnt = styled.div`
  ${mediaTablet} {
    display: grid;
    grid-template-columns: 20% 80%;
    overflow: hidden;
  }
`;

const BeerDetailsStyle = styled.div`
  ${mediaTablet} {
    margin-left: 20px;
  }
`;

const ImgWrapper = styled.div`
  height: 220px;
  margin-bottom: 10px;
  
  ${mediaTablet} {
    margin-bottom: 0;
    height: 70%;
  }
`;

const Name = styled.h2`
  font-size: 24px;
  font-weight: bold;
  margin: 0;
`;

const Tagline = styled.p`
  font-weight: lighter;
  margin-bottom: 30px;
  position: relative;
  
  :before {
    content: '';
    display: block;
    position: absolute;
    bottom: -10px;
    left: 0;
    height: 6px;
    width: 65px;
    background: ${({ theme }) => theme.color.title}
  }
`;

const BeerAttributes = styled.div`
  margin-bottom: 10px;
  
  ${mediaTablet} {
    display: flex;
  }
`;

const BeerAttribute = styled.h4`
  font-weight: bold;
  
  ${mediaTablet} {
    margin-right: 20px;
  }
  
  span {
    font-weight: normal;
  }
`;

const Description = styled.p`
  margin-bottom: 10px;
`;


const BeerDetails = ({
  image_url, name, tagline, ibu, abv, ebc, description,
}) => (
  <BeerDetailsCnt>
    <ImgWrapper>
      <Image src={image_url} alt={name} />
    </ImgWrapper>
    <BeerDetailsStyle>
      <Name>{name}</Name>
      <Tagline>{tagline}</Tagline>
      <BeerAttributes>
        <BeerAttribute>
                IBU:
          {' '}
          <span>{ibu || 0}</span>
        </BeerAttribute>
        <BeerAttribute>
                ABV:
          {' '}
          <span>{abv || 0}</span>
        </BeerAttribute>
        <BeerAttribute>
                EBC:
          {' '}
          <span>{ebc || 0}</span>
        </BeerAttribute>
      </BeerAttributes>
      <Description>{description}</Description>
    </BeerDetailsStyle>
  </BeerDetailsCnt>

);

BeerDetails.defaultProps = {
  ibu: 0,
  abv: 0,
  ebc: 0,
};


BeerDetails.propTypes = {
  image_url: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
  tagline: PropTypes.string.isRequired,
  ibu: PropTypes.number,
  abv: PropTypes.number,
  ebc: PropTypes.number,
  description: PropTypes.string.isRequired,
};


export default BeerDetails;
