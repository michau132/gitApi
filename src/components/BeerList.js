import React, { Fragment } from 'react';
import styled from 'styled-components';
import { compose } from 'recompose';
import PropTypes from 'prop-types';
import withLoading from '../hoc/withLoading';
import withError from '../hoc/withError';
import BeerListItem from './BeerListItem';
import LastBeer from '../hoc/lastBeer';

const BeerListStyle = styled.ul`
  display: flex;
  justify-content: center;
  flex-wrap: wrap;
  padding: 0;
  list-style: none;
  padding-bottom: 150px;
  margin-bottom: 30px;
`;

const BeerList = ({ beerList, handleClick }) => (
  <BeerListStyle>
    {
      beerList.map(beer => (
        <Fragment key={beer.id}>
          <BeerListItem
            handleClick={handleClick}
            {...beer}
          />
        </Fragment>
      ))
    }
  </BeerListStyle>
);

BeerList.propTypes = {
  beerList: PropTypes.arrayOf(PropTypes.shape({
    image_url: PropTypes.string.isRequired,
    name: PropTypes.string.isRequired,
    tagline: PropTypes.string.isRequired,
  }).isRequired).isRequired,
  handleClick: PropTypes.func.isRequired,
};

export default compose(
  withLoading,
  withError,
  LastBeer,
)(BeerList);
