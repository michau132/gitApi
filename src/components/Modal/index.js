import React, { Fragment } from 'react';
import PropTypes from 'prop-types';
import { compose } from 'recompose';
import CloseBtn from '../CloseBtn';
import BeerDetails from './BeerDetails';
import SimilarBeerList from './SimilarBeerList';
import FoodPairList from './FoodPairList';
import withLoading from '../../hoc/withLoading';
import withError from '../../hoc/withError';


export const Modal = ({ beer, similarBeers }) => (
  <Fragment>
    <CloseBtn />
    <BeerDetails {...beer} />
    <FoodPairList food_pairing={beer.food_pairing} />
    <SimilarBeerList similarBeers={similarBeers} />
  </Fragment>
);


Modal.propTypes = {
  beer: PropTypes.shape({
    image_url: PropTypes.string.isRequired,
    name: PropTypes.string.isRequired,
    tagline: PropTypes.string.isRequired,
    ibu: PropTypes.number,
    abv: PropTypes.number,
    ebc: PropTypes.number,
    description: PropTypes.string.isRequired,
    food_pairing: PropTypes.arrayOf(PropTypes.string).isRequired,
  }).isRequired,
  similarBeers: PropTypes.arrayOf(PropTypes.shape({
    id: PropTypes.number.isRequired,
    image_url: PropTypes.string.isRequired,
    name: PropTypes.string.isRequired,
  })).isRequired,
};


export default compose(withLoading, withError)(Modal);
