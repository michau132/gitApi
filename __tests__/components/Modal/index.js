import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { Modal } from '../../../src/components/Modal';

const props = {
  beer: {
    id: 1,
    image_url: 'https://via.placeholder.com/150',
    name: 'Beer',
    tagline: 'Drink or die',
    ibu: 12,
    abv: 6,
    ebc: 32,
    description: 'Some description',
    food_pairing: ['meat', 'vegetables'],
  },
  similarBeers: [
    {
      id: 1,
      image_url: 'https://via.placeholder.com/150',
      name: 'Beer',
    },
    {
      id: 2,
      image_url: 'https://via.placeholder.com/300',
      name: 'Second One',
    },
  ],
  loading: false,
  theme: {
    media: {
      desktop: '15px',
    },
  },
};

describe('<Modal /> component', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<Modal {...props} />);
  });

  test('renders without crashing', () => {
    expect(wrapper).toBeDefined();
  });

  test('matching snapshot', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  test('renders <CloseBtn/>', () => {
    expect(wrapper.find('CloseBtn')).toHaveLength(1);
  });

  test('renders <FoodPairList/>', () => {
    const FoodPairList = wrapper.find('FoodPairList');
    expect(FoodPairList).toHaveLength(1);
    expect(FoodPairList.prop('food_pairing')).toEqual(props.beer.food_pairing);
  });
});
