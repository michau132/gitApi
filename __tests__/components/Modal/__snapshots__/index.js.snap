// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Modal /> component matching snapshot 1`] = `
<Fragment>
  <CloseBtn />
  <BeerDetails
    abv={6}
    description="Some description"
    ebc={32}
    food_pairing={
      Array [
        "meat",
        "vegetables",
      ]
    }
    ibu={12}
    id={1}
    image_url="https://via.placeholder.com/150"
    name="Beer"
    tagline="Drink or die"
  />
  <FoodPairList
    food_pairing={
      Array [
        "meat",
        "vegetables",
      ]
    }
  />
  <SimilarBeerList
    similarBeers={
      Array [
        Object {
          "id": 1,
          "image_url": "https://via.placeholder.com/150",
          "name": "Beer",
        },
        Object {
          "id": 2,
          "image_url": "https://via.placeholder.com/300",
          "name": "Second One",
        },
      ]
    }
  />
</Fragment>
`;
